libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -232,7 +232,9 @@ class MapboxNavigation(
             logger
         )
         routeRefreshController = RouteRefreshController(directionsSession, tripSession, logger)
-        routeRefreshController.start()
+        if (navigationOptions.isRouteRefreshEnabled) {
```

Comment:
add test for that pls

Reply:
Yeah, the problem here is that `routeRefreshController` created here in `init` section. So it is impossible to mock it with current implementation. So the options here are:
- Add `TestOnly` constructor with passing `RouteRefreshController`
- Use reflexion to change instance of rerouteRefreshController by our mocked instance (Yury's idea, I don't know if it will work)

wdyt? @LukasPaczos @RingerJK @Guardiola31337

Reply:
Not sure that second variant is an option, because can replace instance of object which already was called via `start()`

Reply:
@Lebedinsky you can create an `internal object` called something like `RouteRefreshControllerProvider` which returns you an instance of the `RouteRefreshController` instead of calling the constructor in `init`. You can then use `mockkObject` to mock the instance and verify what needs to be verified with no additional `TestOnly` constructors or reflection.

Reply:
@LukasPaczos Awesome, thanks Lukas! I was wondered if we can change current MapboxNavigation implementation) But now I got it, thanks
@RingerJK Thanks